### PR TITLE
Extract CLI string into resource bundle

### LINKED ISSUE #3495: Extract user-facing text to strings file(s)
## Description

Currently, user-facing text in error messages, CLI help and parameter descriptions are embedded in various places in the Java code and elsewhere, which makes it difficult to find and review.

## Possible Solution

If these strings were extracted to dedicated .properties files (or another suitable format), it would make it much easier to review them all at once to ensure consistency and align the tone of messages that were created years apart by different authors.

_(This would also provide a foundation for localization, if that were ever considered for the CLI itself.)_

It would be ideal if all user-facing text could be consolidated in a single format, but given the extensible plug-in–based architecture, and variety of source code formats, I'm not sure that's practical.

As a first step, extracting strings from the core Java code would certainly be a significant improvement.

## Additional Context

Per discussion in https://github.com/dita-ot/dita-ot/pull/3491#discussion_r402084653.

### PR DESCRIPTION
**Pull request by jelovirt**
Extract hard-coded strings from CLI Java code into a resource bundle for easier editing.

Fixes #3495

### CODE REVIEW

**Comment by jelovirt** [comment]
@infotexture Do you think this would be an ok format? JSON or YAML with nested structure instead of period delimited keys would be nice, but this takes less effort to adopt.

**Comment by jelovirt** [comment]
The only thing missing from the property file format is the support for array types. So that you could have an array of strings under one key, for example for usage strings. I need to think about this.

**Comment by jelovirt** [comment]
Reminder to self, add key for command name and generate usages using that and subcommand name, instead of hard coding the whole usage string.

**Comment by infotexture** [comment]
If we think ahead to the idea of one day extracting _all_ user-facing text to strings files (not only from the Java code), other established formats for string resources might be worth considering, such as:

- Apple's [.strings files](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LoadingResources/Strings/Strings.html)
- Android's [strings.xml](https://developer.android.com/guide/topics/resources/string-resource) format
- [.po files](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html) or
- XLIFF

That's beyond the scope of this PR of course, but it's probably worth implementing the initial changes with a view toward future use cases, such as localized toolkit versions. 🔮 

Given the mix of source formats that currently contribute text to the command-line experience (DITA, Java, XML, XSL), I'm not sure how easy it would be to consolidate to a single format for strings, but moving these strings out of the Java code is definitely a step in the right direction. 👍

**Comment by jelovirt** [comment]
Would it be ok if our strings were in some XML format, but then the compilation/integration process would flatten them into a properties file. That would allow runtime to simply read a Java Resource Bundle, but development/authoring experience might be nicer?

**Comment by jelovirt** [comment]
@infotexture could you check the few updates?

File: src/main/resources/cli_en_US.properties [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+install.argument.file=Install plug-in from a local ZIP file
+install.option.force=Force-install plug-in (overwrite existing version)
+# Uninstall subcommand
+uninstall.usage=dita uninstall <id>
+uninstall.argument.id=Uninstall plug-in with the specified ID
+uninstall.error.identifier_not_defined=You must specify plug-in identifier when using the uninstall subcommand
+# Transtypes subcommand
+transtypes.usage=dita transtypes [options]
+# Conversion command
+conversion.usage.input=dita deliverables <file> [options]
```
Comment (jelovirt) [inline src/main/resources/cli_en_US.properties:34]: Because we can't have an array of strings for usage, I used named usage strings.

**Comment by infotexture** [comment]
> could you check the few updates?

Since this was force-pushed with a new single commit, I can't see what's changed since my previous review. 🤔 

I see the result has been merged, but I'll be happy to take another look if that still helps.

**Comment by infotexture** [comment]
> Would it be ok if our strings were in some XML format, but then the compilation/integration process would flatten them into a properties file

That would be fine with me, though we'd probably want to make it clear which format is the source, and which is the generated output, so people don't try to edit the wrong file(s).

**Comment by jelovirt** [comment]
The generated file would not be in Git and these files are not something that are intended to be edited by end users.

**Comment by infotexture** [comment]
Right, I get that, but would the distribution package contain both, or only the generated file?

**Pull request merged**